ui/src/components/admin/Notifications/PublishNotifications.js

```diff
@@ -0,0 +1,337 @@
+
+    if (existingNotification) {
+      existingNotification.clear();
+    }
+  };
+
+  const showUnexpectedPublishError = error => {
+    appendNotification({
+      type: NOTIFICATION_TYPES.ERROR,
+      message: 'Publish Error: An unexpected error has occurred.',
+      details: error?.error?.message || error?.message || error,
+      timeout: false,
+    });
+  };
+
+  const updatePublishNotificationsFromStatus = async () => {
+    const { queue, stopped, success, failed } = publishProgress;
+
+    // Add publish notifications for individual publishes in the queue
+    const individualQueuedPublishes = (queue || []).filter(
+      x => x.publishType === PUBLISH_TYPES.individual,
+    );
+    individualQueuedPublishes.forEach(publishTask => {
+      const modelName = publishTask.modelName;
+      if (!getPublishNotifications().find(x => x.modelName === modelName)) {
+        addPublishNotification(modelName);
+      }
+    });
+
+    // Remove publish notification and show stopped notification for stopped individual publishes
+    const individualStoppedPublishes = (stopped || []).filter(
+      x => x.publishType === PUBLISH_TYPES.individual,
+    );
+    individualStoppedPublishes.forEach(stoppedPublish => {
+      const modelName = stoppedPublish.modelName;
+      removePublishNotification(modelName);
+      showStoppedPublishNotification(modelName);
+    });
+
+    // Remove publish notification and show success notification for completed individual publishes
+    const individualCompletedPublishes = (success || []).filter(
+      x => x.publishType === PUBLISH_TYPES.individual,
+    );
+    individualCompletedPublishes.forEach(completedPublish => {
+      const modelName = completedPublish.modelName;
+      removePublishNotification(modelName);
+      showSuccessfulPublishNotification(modelName);
+    });
+
+    // Remove publish errors for re-queued bulk publishes (re-published)
+    const bulkQueuedPublishes = (queue || []).filter(x => x.publishType === PUBLISH_TYPES.bulk);
+    bulkQueuedPublishes.forEach(queuedPublish => {
+      const modelName = queuedPublish.modelName;
+      hideErrorPublishNotification(modelName);
+    });
+
+    // Remove publish errors for completed bulk publishes
+    const bulkCompletedPublishes = (success || []).filter(
+      x => x.publishType === PUBLISH_TYPES.bulk,
+    );
+    bulkCompletedPublishes.forEach(completedPublish => {
+      const modelName = completedPublish.modelName;
+      hideErrorPublishNotification(modelName);
+    });
+
+    // Add error notifications for failed publishes
+    const failedPublishes = failed || [];
+
+    failedPublishes.forEach(failedPublish => {
+      const modelName = failedPublish.modelName;
+
+      if (failedPublish.publishType === PUBLISH_TYPES.individual) {
+        removePublishNotification(modelName);
+      }
+
+      showErrorPublishNotification(modelName, failedPublish);
```

Comment:
A big difference between publish notifications and import notifications is that publish has no concept of "bulk" errors. For publish, the errors are all equal and we simply show them here, but for import there is a lot of extra logic surrounding whether the import was bulk or not.